interceptor/tracing_interceptor.go

```diff
@@ -886,16 +1009,47 @@ func (t *tracingInterceptor) writeSpanToHeader(span TracerSpan, header map[strin
 	return nil
 }
 
-func newErrFut(ctx workflow.Context, err error) workflow.ChildWorkflowFuture {
+func (t *tracingInterceptor) writeSpanToNexusHeader(span TracerSpan, header nexus.Header) error {
+	// Serialize span to map
+	data, err := t.tracer.MarshalSpan(span)
+	if err != nil || len(data) == 0 {
+		return err
+	}
+	// Put on header
+	for k, v := range data {
+		header.Set(k, v)
+	}
+	return nil
+}
+
+func (t *tracingInterceptor) readSpanFromNexusHeader(header nexus.Header) (TracerSpanRef, error) {
+	return t.tracer.UnmarshalSpan(header)
+}
+
+type errFut struct{ inner workflow.Future }
+
+func newErrFut(ctx workflow.Context, err error) *errFut {
```

Comment:
To be clear my suggestion was keep `newErrFut` returning a type that implements `workflow.ChildWorkflowFuture` and add a new function `newNexusErrFut` that returns a `nexusOperationFuture`

Reply:
can also rename `newErrFut ` to something like `newWorkflowErrFut` , but I have no strong opinion there

Reply:
I simplified this a bit and I think what we have now is a bit cleaner.